pkg/attributes/attributes.go

```diff
@@ -0,0 +1,353 @@
+package attributes
+
+import (
+	"encoding/json"
+	"errors"
+
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// Attributes provides a way to add a map of arbitrary YAML/JSON
+// objects.
+type Attributes map[string]apiext.JSON
+
+// GetDecodedInto allows decoding the attribute with the given key
+// into a given interface. The provided interface should be a pointer
+// to a struct, to an array, or to any simple type.
+//
+// An error is returned if the provided interface type is not compatible
+// with the attribute content
+func (attributes Attributes) GetDecodedInto(key string, into interface{}) error {
```

Comment:
I'd suggest `DecodeKeyInto` as a function name here, or something similar. This name might be too similar to `DecodeInto` below.

Also I think `DecodeInto` and `GetDecodedInto` should be adjacent in this file.